jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaDockerContextGenerator.java

```diff
@@ -282,9 +285,9 @@ public void generate(Path targetDirectory) throws IOException {
         // 'baseExtractionPath' of '/app/classes', and an 'actualExtractionPath' of
         // '/app/classes/com/test/HelloWorld.class', the resolved destination would be
         // 'target/jib-docker-context/classes/com/test/HelloWorld.class'.
-        Path destination =
-            directoryInContext.resolve(
-                Paths.get(copyDirective.extractionPath).relativize(layerEntry.getExtractionPath()));
+        Path baseExtractionPath = Paths.get(copyDirective.extractionPath.toString());
+        Path relativeEntryPath = baseExtractionPath.relativize(layerEntry.getExtractionPath());
```

Comment:
Hmm, maybe we can use `copyDirective.extractionPath` directly and add a `relativize` method to `AbsoluteUnixPath` after we change `LayerEntry#getExtractionPath` to return an `AbsoluteUnixPath`.

Reply:
Oh, good catch.

Reply:
Perhaps we can think about this after #1014 (or in #1014).